mkdocs_literate_nav/plugin.py

```diff
@@ -68,45 +78,46 @@ def on_nav(
             )
 
 
-def resolve_directories_in_nav(
-    nav_data,
-    files: mkdocs.structure.files.Files,
-    nav_file_name: str,
-    implicit_index: bool,
-    markdown_config: dict | None = None,
-):
-    """Walk through a standard MkDocs nav config and replace `directory/` references.
-
-    Directories, if found, are resolved by the rules of literate nav insertion:
-    If it has a literate nav file, that is used. Otherwise an implicit nav is generated.
-    """
-
-    def get_nav_for_dir(path: str) -> tuple[str, str] | None:
-        file = files.get_file_from_path(os.path.join(path, nav_file_name))
-        if not file:
-            return None
-        log.debug(f"Navigation for {path!r} based on {file.src_path!r}.")
-
-        # https://github.com/mkdocs/mkdocs/blob/ff0b726056/mkdocs/structure/nav.py#L113
-        # Prevent the warning in case the user doesn't also end up including this page in
-        # the final nav, maybe they want it only for the purpose of feeding to this plugin.
-        mkdocs.structure.pages.Page(None, file, {})
-
-        # https://github.com/mkdocs/mkdocs/blob/fa5aa4a26e/mkdocs/structure/pages.py#L120
-        with open(file.abs_src_path, encoding="utf-8-sig") as f:
-            return nav_file_name, f.read()
-
-    globber = MkDocsGlobber(files)
-    nav_parser = parser.NavParser(
-        get_nav_for_dir, globber, implicit_index=implicit_index, markdown_config=markdown_config
-    )
-
-    result = None
-    if not nav_data or get_nav_for_dir("."):
-        result = nav_parser.markdown_to_nav()
-    if not result:
-        result = nav_parser.resolve_yaml_nav(nav_data)
-    return result or []
+    def _resolve_directories_in_nav(
```

Comment:
@oprypin these changes were just to expose the `nav_parser` to the plugin by making this a method of the plugin.